### PR TITLE
test(test): add wellknown integration test

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add wellknown api integration test

get sign-in-exp for admin console and general app.
log-3670
log-3671

## Testing

test locally
@logto-io/eng

### CODE REVIEW

File: packages/integration-tests/tests/wellknown.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    const client = new MockClient();
+
+    await client.initSession();
+
+    assert(client.interactionCookie, new Error('Session not found'));
+
+    const response = await getWellKnownSignInExperience(client.interactionCookie);
+
+    // Should support sign-in and register
+    expect(response).toMatchObject({ signInMode: 'SignInAndRegister' });
```
Comment (IceHe) [inline packages/integration-tests/tests/wellknown.test.ts:38]: nit: use `toHaveProperty` to match only one property instead?

File: packages/integration-tests/tests/wellknown.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { getWellKnownSignInExperience } from '@/api';
+import MockClient from '@/client';
+import { adminConsoleRedirectUri } from '@/constants';
+
+describe('wellknown api', () => {
+  it('get /.well-known/sign-in-exp for AC', async () => {
+    const client = new MockClient({ appId: adminConsoleApplicationId });
+    await client.initSession(adminConsoleRedirectUri);
+
+    assert(client.interactionCookie, new Error('Session not found'));
```
Comment (IceHe) [inline packages/integration-tests/tests/wellknown.test.ts:13]: just curious why not use `expect(…).toMatchError(…)`?

Comment (simeng-li) [reply packages/integration-tests/tests/wellknown.test.ts:13]: not for test use. interactionCookie should not be empty here, just use it as a type guard.

File: packages/integration-tests/tests/wellknown.test.ts (hunk 1 above)
Comment (simeng-li) [reply packages/integration-tests/tests/wellknown.test.ts:38]: Feels like not much difference.

File: packages/integration-tests/src/client/index.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default class MockClient {
   private readonly storage: MemoryStorage;
   private readonly logto: LogtoClient;
 
-  constructor(config?: LogtoConfig) {
+  constructor(config?: Partial<LogtoConfig>) {
```
Comment (darcyYe) [inline packages/integration-tests/src/client/index.ts:25]: Why change to `Partial` here?
Does `defaultConfig` cover all required props a client needs?

File: packages/integration-tests/tests/wellknown.test.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    const response = await getWellKnownSignInExperience(client.interactionCookie);
+
+    expect(response).toMatchObject({
+      signInMethods: {
+        username: 'primary',
+        email: 'disabled',
+        sms: 'disabled',
+        social: 'disabled',
+      },
+      signInMode: 'SignIn',
```
Comment (darcyYe) [inline packages/integration-tests/tests/wellknown.test.ts:24]: I'm curious about how you get these values. Are they default SIE of an arbitrary new client?

File: packages/integration-tests/src/client/index.ts (hunk 3 above)
Comment (simeng-li) [reply packages/integration-tests/src/client/index.ts:25]: Yep

File: packages/integration-tests/tests/wellknown.test.ts (hunk 4 above)
Comment (simeng-li) [reply packages/integration-tests/tests/wellknown.test.ts:24]: Hard coded for the AC application SIE.

**Review by darcyYe — APPROVED** [review]
lgtm

**Pull request merged**